Reduce yaml files for installer_extended

hjluo opened the pull request:
##
### Reduce yaml files for installer_extended
- Related ticket: 
   -  https://progress.opensuse.org/issues/163226
- Related MR:
  -  https://gitlab.suse.de/qe-yam/openqa-job-groups/-/merge_requests/315
- Needles: 
    - N/A
- Verification run:  
   - [**VR**](https://openqa.suse.de/tests/overview?build=hjluo%2Fos-autoinst-distri-opensuse%23reduce_installer_extended&version=15-SP7&distri=sle)

jknphy reviewed (commented):
we also need verification for the textmode one.
Could you please summarize the differences you found that stop you to unify it, I can see for example that
installation/validate_beta_message or switch_keyboard_layout were not taken into account.

hjluo commented:
> we also need verification for the textmode one. Could you please summarize the differences you found that stop you to unify it, I can see for example that installation/validate_beta_message or switch_keyboard_layout were not taken into account.

- Added x86_64 textmode
- Added back installation/validate_beta_message
- for switch_keyboard_layout, [**it's not work for s390x**](https://openqa.suse.de/tests/15252897#step/switch_keyboard_layout/2).

rakoenig reviewed (approved):
LGTM.

jknphy commented:
> > we also need verification for the textmode one. Could you please summarize the differences you found that stop you to unify it, I can see for example that installation/validate_beta_message or switch_keyboard_layout were not taken into account.
> 
> * Added x86_64 textmode
> * Added back installation/validate_beta_message
> * for switch_keyboard_layout, [**it's not work for s390x**](https://openqa.suse.de/tests/15252897#step/switch_keyboard_layout/2).

I remember that is not a bug in s390x (don't remember why...though, but I think it is related how openQA does that typing), but the solution is not removing the test where it works (these extended test already had features that were more spread in the test coverage and we condesate here, so it is crucial not drop anything), we should just adapt the test module to return if s390x for example with a record_info about the issue (you might need to search in history of the test module to know why that happens, it might be there, otherwise a generic message would help).
Is there anything else which cannot be unified? if not, we can have it in test_cases folder.

hjluo commented:
> > > we also need verification for the textmode one. Could you please summarize the differences you found that stop you to unify it, I can see for example that installation/validate_beta_message or switch_keyboard_layout were not taken into account.
> > 
> > 
> > 
> > * Added x86_64 textmode
> > * Added back installation/validate_beta_message
> > * for switch_keyboard_layout, [**it's not work for s390x**](https://openqa.suse.de/tests/15252897#step/switch_keyboard_layout/2).
> 
> I remember that is not a bug in s390x (don't remember why...though, but I think it is related how openQA does that typing), but the solution is not removing the test where it works (these extended test already had features that were more spread in the test coverage and we condesate here, so it is crucial not drop anything), we should just adapt the test module to return if s390x for example with a record_info about the issue (you might need to search in history of the test module to know why that happens, it might be there, otherwise a generic message would help). Is there anything else which cannot be unified? if not, we can have it in test_cases folder.

I added a keyboard key in the default for aarch64, ppc64le and x86_64 to solve this issue.

hjluo commented:
> > > > > we also need verification for the textmode one. Could you please summarize the differences you found that stop you to unify it, I can see for example that installation/validate_beta_message or switch_keyboard_layout were not taken into account.
> > > > 
> > > > 
> > > > 
> > > > * Added x86_64 textmode
> > > > * Added back installation/validate_beta_message
> > > > * for switch_keyboard_layout, [**it's not work for s390x**](https://openqa.suse.de/tests/15252897#step/switch_keyboard_layout/2).
> > > 
> > > 
> > > I remember that is not a bug in s390x (don't remember why...though, but I think it is related how openQA does that typing), but the solution is not removing the test where it works (these extended test already had features that were more spread in the test coverage and we condesate here, so it is crucial not drop anything), we should just adapt the test module to return if s390x for example with a record_info about the issue (you might need to search in history of the test module to know why that happens, it might be there, otherwise a generic message would help). Is there anything else which cannot be unified? if not, we can have it in test_cases folder.
> > 
> > 
> > I added a key called keyboard in the default for aarch64, ppc64le and x86_64 to solve this issue.
> 
> doesn't solve the issue completely, because we still need to unify the schedules, you can do it editing the module as I mentioned.

Now changed switch_keyboard_layout to run on qemu and rerun all VR, please take a look, thanks

jknphy commented on tests/installation/language_keyboard/switch_keyboard_layout.pm:
```diff
@@ -8,8 +8,14 @@ use base 'y2_installbase';
+use Utils::Backends;
 
 sub run {
+    if (!is_qemu) {
```
`unless (is_qemu)` usually nicer (specially when there is single condition, but not big preference.

hjluo replied:
changed.

jknphy commented on tests/installation/language_keyboard/switch_keyboard_layout.pm:
```diff
@@ -8,8 +8,14 @@ use base 'y2_installbase';
 
 sub run {
+    if (!is_qemu) {
+        record_info('Skip', 'This module only run on installations not using remote protocol');
```
s/This module only run on installations not using remote protocol/Switch keyboard not testeable for non-qemu scenarios/
(I would recommend you this above as I don't know what you are referring with remote protocol, I guess you mean remote workers).

jknphy commented on schedule/yast/installer_extended/installer_extended.yaml:
```diff
@@ -7,90 +7,24 @@ description: >
+  access_beta:
+    - installation/validate_beta_message
+    - installation/access_beta_distribution
+    - installation/language_keyboard/switch_keyboard_layout
```
if now finally is unified and we have only single file for all the architectures, you can move it to test_cases/

hjluo replied:
changed.

jknphy reviewed (approved):
LGTM, other than putting the file in the right path.

jknphy commented on schedule/yam/test_cases/installer_extended.yaml:
```diff
@@ -7,13 +7,11 @@ description: >
+  access_beta:
+    - installation/validate_beta_message
+    - installation/access_beta_distribution
+    - installation/language_keyboard/switch_keyboard_layout
```
one small detail, this belongs to product selection not to beta.

hjluo replied:
> one small detail, this belongs to product selection not to beta.

That's the tricky part, it we set
